### PR TITLE
Refactor Code to use Boost instead of CXX

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- Refactored code to use Boost Test instead to fix compilation issues for timer test.

### CODE REVIEW

File: test/timer/timer_test.cxx [hunk 1]
```diff
@@ -37,4 +37,4 @@ BOOST_AUTO_TEST_CASE(test_timer) {
 BOOST_AUTO_TEST_CASE(test_lowres_timer) {
   timer_test::test_lowres_timer();
 }
-BOOST_AUTO_TEST_SUITE_END()
+BOOST_AUTO_TEST_SUITE_END()
```
Comment (shantanuchhabra) [inline test/timer/timer_test.cxx]: Let's add the newline back.

**Comment by nickjong** [comment]
What's the relationship between these changes and #2478? @hoytak 

Any reason not just to change the definition of the macro, instead of the call sites?

File: src/toolkits/style_transfer/style_transfer.hpp [hunk 2]
```diff
@@ -36,6 +36,8 @@ class EXPORT style_transfer : public ml_model_base {
   void train(gl_sarray style, gl_sarray content,
              std::map<std::string, flexible_type> opts);
 
+  gl_sarray predict(variant_type data, std::map<std::string, flexible_type> opts);
```
Comment (hoytak) [inline src/toolkits/style_transfer/style_transfer.hpp]: I think this is the only actual change here...

**Comment by abhishekpratapa** [comment]
> What's the relationship between these changes and #2478? @hoytak
> 
> Any reason not just to change the definition of the macro, instead of the call sites?

Genius. Yeah.

**Pull request closed without merging**